cogs/birthday/birthday.py

```diff
@@ -433,12 +433,49 @@ async def getSelfBirthday(self, ctx: Context):
                 if month and day:
                     birthday = date(2020, month, day)
                     birthdayStr = "{0:%B} {0:%d}".format(birthday)
-                    await ctx.author.send(
-                        f"{headerGood}: Your birthday is "
-                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-                    )
-                    return
-
+                    try:
+                        await ctx.author.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                        )
+                        return
+                    except discord.Forbidden:
+
+                        def check(msg: discord.Message):
+                            return msg.author == ctx.author and msg.channel == ctx.channel
+
+                        try:
+                            await ctx.send(
+                                "You have disabled DMs from this server. Would you still like to continue here? "
+                                "The your birthday will be sent here and deleted after a short delay. "
+                                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
+                            )
+                            response = await self.bot.wait_for(
+                                "message", timeout=30.0, check=check
+                            )
+                        except asyncio.TimeoutError:
+                            await ctx.send(f"{headerBad}: No response detected. Aborting.")
+                            return
+
+                        if response.content.lower() != "yes":
+                            await ctx.send("Alright then")
+                            return
+
+                        can_delete = ctx.channel.permissions_for(ctx.me).manage_messages
+                        if can_delete:
+                            await ctx.send(
+                                f"{headerGood}: Your birthday is "
+                                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}.",
+                                delete_after=5,
+                            )
+                            return
+                        else:
+                            await ctx.send(
+                                f"{headerBad}: Unable to delete messages. Please "
+                                "notify the server administrators to give me the "
+                                '"Manage Messages" permission'
```

Comment:
I don't think users need to know about the "Manage Messages" permission. Can just tell them to contact a server admin.
```suggestion
                                f"{headerBad}: Unable to delete messages. Please "
                                "notify a server administrator."
```

Reply:
Permissions check has been removed entirely.